packages/edit-post/src/components/secondary-sidebar/list-view-sidebar.js

```diff
@@ -29,20 +33,42 @@ import ListViewOutline from './list-view-outline';
 
 export default function ListViewSidebar() {
 	const { setIsListViewOpened } = useDispatch( editPostStore );
+	const { clearSelectedBlock } = useDispatch( blockEditorStore );
+	const { hasBlockSelection } = useSelect(
+		( select ) => ( {
+			hasBlockSelection:
+				!! select( blockEditorStore ).getBlockSelectionStart(),
+		} ),
+		[]
+	);
 
 	const focusOnMountRef = useFocusOnMount( 'firstElement' );
 	const headerFocusReturnRef = useFocusReturn();
 	const contentFocusReturnRef = useFocusReturn();
 
+	const [ tab, setTab ] = useState( 'list-view' );
+
 	function closeOnEscape( event ) {
+		// If there is a block selection, then skip closing the list view
+		// and clear out the block selection instead.
+		if (
+			tab === 'list-view' &&
+			event.keyCode === ESCAPE &&
+			! event.defaultPrevented &&
+			hasBlockSelection
+		) {
+			event.preventDefault();
+			clearSelectedBlock();
+			speak( __( 'All blocks deselected.' ), 'assertive' );
+			return;
+		}
```

Comment:
Something I noticed is that if you have focus on the List View tab button or the close button for the sidebar and press escape, it still deselects all blocks. 

I wondered if it perhaps only makes sense to deselect blocks when focus is within the ListView component itself (i.e. on a block in List View). What do you think @andrewserong  @alexstine?

Reply:
Oh, great point. Yes, I think it makes more sense to only deselect while focus is _within_ the list view (i.e. on a block as you mention). I'll be AFK from tomorrow through Monday, so can have a go at updating the logic on Tuesday if I run out of time today. Keen to hear what Alex thinks, though.

Reply:
Agreed. 👍